client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -1,20 +1,46 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import {
   DatePicker,
   DatePickerProps,
 } from '@mui/x-date-pickers/DatePicker/DatePicker';
 import { BasicTextInput } from '../../TextInput/BasicTextInput';
 import { StandardTextFieldProps, TextFieldProps } from '@mui/material';
 import { useAppTheme } from '@common/styles';
+import { DateUtils } from '@common/intl';
+import { useDebounceCallback } from '@common/hooks';
 
 export const BaseDatePickerInput: FC<
-  Omit<DatePickerProps<Date>, 'renderInput'>
-> = props => {
+  Omit<DatePickerProps<Date>, 'renderInput' | 'value'> & {
+    onChange(date: Date): void;
+    value: Date | string | null;
+  }
+> = ({ disabled, onChange, value, ...props }) => {
   const theme = useAppTheme();
+  const [internalValue, setInternalValue] = useState<Date | null>(null);
+
+  useEffect(() => {
+    // This sets the internal state from parent when first loading (i.e. when
+    // the internal date is still empty)
+    if (value && internalValue === null)
+      setInternalValue(DateUtils.getDateOrNull(value));
+  }, [value]);
```

Comment:
`internalValue` is missing from the dependency array...

Reply:
Yes, that is correct. If we set `internalValue` on change of `internalValue`, we'd have an infinite loop. ;)

Reply:
You should always include the full dependencies array if you want to avoid subtle bugs (see the pitfall box in the useEffect docs).

Why is there a infinite loop? the `if` should catch it shouldn't it?

Reply:
Yes, I have read about that and seen the ESLint warnings. However, I have yet to see an example of these "subtle bugs" that has convinced me to over-stuff the dependency array. In this case, why would we want to re-run the useEffect function on change of `internalValue`? All it does is (possibly) _update_ `internalValue`? And we're comparing against the initial state of `internalValue`, so it'll always be `null` on first load.

I'm open to being convinced otherwise, I just don't really see a problem in this case. To me it just adds confusion.

Reply:
beside its a React convention/contract and the occasional bug caused by not specifying it, it becomes IMHO also harder to read. For example, in my previous comment I was not sure if the comment was wrong or the dependency was missing intentionally. 

When you not specify the full array your code become more "concurrent" because you have to mentally keep track which state variables are in and in what order things are called. If you specify all dependencies in the array you can just systematically handle all possible combinations in useEffect which simplifies the problem a bit...